### PR TITLE
feat(curriculum):  add Data Structures Quiz

### LINKED ISSUE #56219: Data Structures

### PR DESCRIPTION
**Pull request by devshah207**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56219 

I added data structures quiz in this issue.

### CODE REVIEW

**Comment by ilenia-magoni** [comment]
the lint is failing, can you check that out and fix it? If you have any questions, please reach out in our [Discord community](https://chat.freecodecamp.org).

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-data-structures/66edd5583761e4f6dae19606.md [hunk 1]
```diff
@@ -17,439 +17,440 @@ Answer all of the questions below correctly to pass the quiz.
...
-Placeholder answer
+O(1) amortized, but O(n) when resizing occurs
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following is a property of a binary min-heap?
+
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-data-structures/66edd5583761e4f6dae19606.md:65]:
```suggestion
Which of the following is a property of a binary min-heap?
```

Extra line break

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-data-structures/66edd5583761e4f6dae19606.md [hunk 2]
```diff
@@ -17,439 +17,440 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+Root contains the smallest element
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Choose the coorect statement about hash collisions in a hash map?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-data-structures/66edd5583761e4f6dae19606.md:87]:
```suggestion
Choose the correct statement about hash collisions in a hash map.
```

Not a question, so should end with a full-stop. Also, a typo.

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-data-structures/66edd5583761e4f6dae19606.md [hunk 3]
```diff
@@ -17,439 +17,440 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+Heap
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+In a doubly linked list, each node contains ?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-data-structures/66edd5583761e4f6dae19606.md:175]:
```suggestion
In a doubly linked list, what does each node contain?
```

Grammar nit.

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-data-structures/66edd5583761e4f6dae19606.md [hunk 4]
```diff
@@ -17,439 +17,440 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+Data and two pointers (next and previous)
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+The value of each node in a max-heap is?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-data-structures/66edd5583761e4f6dae19606.md:197]:
```suggestion
The value of each node in a max-heap is:
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-data-structures/66edd5583761e4f6dae19606.md [hunk 5]
```diff
@@ -17,439 +17,440 @@ Answer all of the questions below correctly to pass the quiz.
...
 ---
 
-Placeholder distractor 2
+Each node has two pointers
 
 ---
 
-Placeholder distractor 3
+The list is circular by default
+
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-data-structures/66edd5583761e4f6dae19606.md:232]:
```suggestion
The list is circular by default
```

Extra new line.

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-data-structures/66edd5583761e4f6dae19606.md [hunk 6]
```diff
@@ -17,439 +17,440 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --text--
 
-Placeholder question
+What is the time complexity of accessing an element in a static array?
 
 #### --distractors--
 
-Placeholder distractor 1
+O( log n)
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-data-structures/66edd5583761e4f6dae19606.md:312]:
```suggestion
O(log n)
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-data-structures/66edd5583761e4f6dae19606.md [hunk 7]
```diff
@@ -17,439 +17,440 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+Does not allow duplicate elements
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What differs a dynamic array from a static array?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-data-structures/66edd5583761e4f6dae19606.md:374]:
```suggestion
What differentiates a dynamic array from a static array?
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-data-structures/66edd5583761e4f6dae19606.md [hunk 8]
```diff
@@ -17,439 +17,440 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+Static arrays are only used in Java
 
 #### --answer--
 
-Placeholder answer
+Dynamic arrays can grow whereas Static arrays have fixed size
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-data-structures/66edd5583761e4f6dae19606.md:390]:
```suggestion
Dynamic arrays can grow whereas static arrays have fixed size 
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Some formatting and grammar nits, but you are off to a great start!

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-data-structures/66edd5583761e4f6dae19606.md (hunk 8 above)
Comment (devshah207) [reply curriculum/challenges/english/25-front-end-development/quiz-javascript-data-structures/66edd5583761e4f6dae19606.md]: Hey!! A big thanks for giving me guidance and sorry for the mistakes!!

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-data-structures/66edd5583761e4f6dae19606.md [hunk 9]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 ---
 
-Placeholder distractor 3
+Unsorted Elements
 
 #### --answer--
 
-Placeholder answer
-
+Fixed Size
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-data-structures/66edd5583761e4f6dae19606.md:454]:
```suggestion
Fixed Size

```

you have accidentally removed a line here

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-data-structures/66edd5583761e4f6dae19606.md [hunk 10]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --distractors--
 
-Placeholder distractor 1
+Stack allows random access while queue does not
 
 ---
 
-Placeholder distractor 2
+Queue follows LIFO and stack follows FIFO
```
Comment (gikf) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-data-structures/66edd5583761e4f6dae19606.md:292]:
```suggestion
Queue follows "Last in, first out" (LIFO) and stack follows "First in, first out" (FIFO)
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-data-structures/66edd5583761e4f6dae19606.md [hunk 11]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+Both follow the same order of operations
 
 #### --answer--
 
-Placeholder answer
+Stack follows LIFO and queue follows FIFO
```
Comment (gikf) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-data-structures/66edd5583761e4f6dae19606.md:300]:
```suggestion
Stack follows "Last in, first out" (LIFO) and queue follows "First in, first out" (FIFO)
```

**Comment by gikf** [comment]
Hey @devshah207

did you get a chance to look at the suggested changes?

**Review by naomi-lgbt — APPROVED** [review]

**Review by gikf — APPROVED** [review]

**Comment by gikf** [comment]
Hi @devshah207. Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 📝

**Pull request merged**